mcan-core/src/lib.rs

```diff
@@ -262,6 +266,14 @@ pub unsafe trait CanId {
 ///
 /// [`mcan`]: <https://docs.rs/crate/mcan/>
 pub unsafe trait Dependencies<Id: CanId> {
+    /// Pointer to the `Message RAM`.
+    ///
+    /// Only 2 most significant bytes are relevant.
+    ///
+    /// MCAN uses 16-bit addressing internally. In order to validate the
+    /// correctness of the `Message RAM` placement, target HAL has to provide
+    /// the information about its absolute position in RAM.
+    fn message_ram(&self) -> *const ();
```

Comment:
I always thought of message RAM as the memory that the application allocates for the peripheral to use. This is consistent with the last paragraph of the doc comment, but the first line seems to make "message RAM" out to be the whole block of addressable memory. I'd like to make the distinction clearer. Here's one option:

```suggestion
    /// Pointer to the start of memory that can be used for `Message RAM`.
    ///
    /// Only 2 most significant bytes are relevant.
    ///
    /// MCAN uses 16-bit addressing internally. In order to validate the
    /// correctness of the `Message RAM` placement, the target HAL has to
    /// provide information about its absolute position in RAM.
    fn eligible_message_ram_start(&self) -> *const ();
```